### ISSUE TITLE
Support resolving schemas by content hash in 

### ISSUE THREAD

**Issue by gr8routdoors** [Apicurio/apicurio-registry#3237]
### Feature or Problem Description

As a registry client, it would be nice to be able to resolve schemas with some type of ID that does not vary between Registry instances (i.e. different environments in a CD pipeline).  Apicurio already has a [getContentByHash endpoint](https://www.apicur.io/registry/docs/apicurio-registry/2.4.x/assets-attachments/registry-rest-api.htm#tag/Artifacts/operation/getContentByHash) to look up schema content by its SHA256 sum.  This is also already supported in the [RegistryClient](https://github.com/Apicurio/apicurio-registry/blob/main/client/src/main/java/io/apicurio/registry/rest/client/RegistryClient.java#L172).  However, it's missing from the `DefaultSchemaResolver` (see [here](https://github.com/Apicurio/apicurio-registry/blob/c712eca2f6dc128616f10d238169eb5c00a8102e/schema-resolver/src/main/java/io/apicurio/registry/resolver/DefaultSchemaResolver.java#L151-L168) and [here](https://github.com/Apicurio/apicurio-registry/blob/c712eca2f6dc128616f10d238169eb5c00a8102e/schema-resolver/src/main/java/io/apicurio/registry/resolver/DefaultSchemaResolver.java#L93-L102)).  This issue is to fill in that functionality so that things like Kafka Serde that use SchemaResolver can benefit from this capability. 

### Proposed Solution

1. Update `DefaultSchemaResolver` so that it can support resolution of artifacts by their content hash
2. Allow for the Content Hash header name to be configurable in [SchemaResolverConfig](https://github.com/Apicurio/apicurio-registry/blob/c712eca2f6dc128616f10d238169eb5c00a8102e/schema-resolver/src/main/java/io/apicurio/registry/resolver/SchemaResolverConfig.java), just like all the rest of the headers already supported for schema resolution.

**Comment by gr8routdoors** [comment]
I'm happy to take a stab at implementing this if the team would welcome it.

**Comment by EricWittmann** [comment]
A contribution for this would be very welcome.  It's funny because we were just discussing this feature (content hash) recently, and we were wondering whether anyone was using it. :)

The only thing I would make sure to mention to you is that the content hash feature **might** be deprecated in the future **IF** we allow client-provided IDs (contentId, globalId).  Why might we do that?  To support a GitOps workflow for managing registry content.  This is undecided, but if the client were able to provide the `contentId`, then the hash would be no longer useful (we think).

**Comment by gr8routdoors** [comment]
Fantastic.  Funny thing about us wanting to go with `contentHash` is that we wanted an ID that the client can generate, so moving to client-provided IDs in the future would be just fine.  However, it would be nice if there was a string option for client-provided IDs, so we could use something like a GUID to simplify uniqueness guarantees.

**Comment by EricWittmann** [comment]
Good point - I think it's likely that when we allow user-provided IDs they would be strings.  They are numbers right now because we use DB counters/sequences.

**Comment by carlesarnal** [comment]
Closing as the related PR has been merged.
